ha/setup/create_cluster.py

```diff
@@ -0,0 +1,126 @@
+#!/bin/python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import argparse
+from ha.execute import SimpleCommand
+from cortx.utils.log import Log
+
+class ClusterCreateError(Exception):
+
+    """Exception to indicate that cluster can't be created due to some error."""
+
+class ClusterAuthError(ClusterCreateError):
+
+    """Exception to indicate that authorization procedure failed."""
+
+class ClusterSetupError(ClusterCreateError):
+
+    """Exception to indicate that cluster setup procedure failed."""
+
+
+def cluster_auth(username, password, nodelist):
+    """Authorizes cluster nodes.
+
+    Parameters:
+            username - just a user name to make cluster authorization
+            password - just a password for aforementioned user
+            nodelist - List with nodes which to authorize
+
+    Returns: None.
+
+    Exceptions:
+        ClusterCreateError: generic exception to catch all creation-related
+        problems.
+        ClusterAuthError: failure happened during setup operation.
+    """
+    nodes = " ".join(nodelist)
+    cmd_auth = f"pcs cluster auth -u {username} -p {password} {nodes}"
+    try:
+        SimpleCommand().run_cmd(cmd_auth)
+    except Exception:
+        raise ClusterAuthError("Failed to authorize the node")
```

Comment:
It will be good to have nodelist in exception message.

Reply:
Added.